File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    // Copy linked list to temp variable to traverse to find the length
    temp = head
    let count = 0
    // Find length of linked list
    while(temp.next != null){
        count += 1
        temp = temp.next
    }
    
    // Find middle index of linked list
    count /= 2
    
    // Traverse linked list to the middle node and return
    while(count > 0){
        count -= 1
        head = head.next
    }
    return head
};

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def middleNode(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        # Find the length of the linked list
        temp = head
        count = 0
        while(temp):
            count += 1
            temp = temp.next

        count /= 2
        # Go to the middle and return the head
        while(count > 0):
            count -= 1
            head = head.next
        
        return head

        